lxd/container.go

```diff
@@ -748,22 +748,24 @@ func containerCreateInternal(d *Daemon, args containerArgs) (container, error) {
 
 func containerConfigureInternal(c container) error {
 	// Find the root device
-	for _, m := range c.ExpandedDevices() {
-		if m["type"] != "disk" || m["path"] != "/" || m["size"] == "" {
-			continue
-		}
+	localDevices := c.LocalDevices()
+	_, rootDiskDevice := containerGetRootDiskDevice(localDevices)
+	if rootDiskDevice["pool"] == "" {
+		expandedDevices := c.ExpandedDevices()
+		_, rootDiskDevice = containerGetRootDiskDevice(expandedDevices)
+	}
```

Comment:
Hmm, what?

_, rootDiskDevice := containerGetRootDiskDevice(c.ExpandedDevices())

Should do exactly that right?